kaolin/graphics/SoftRenderer.py

```diff
@@ -387,37 +402,20 @@ def rasterize(self, vertices, faces, textures):
 
         """
 
-        faces = self.vertices_to_faces(vertices, faces)
-
-        # If mode is unspecified, render rgb, depth, and alpha channels
-        if mode is None:
-            out = kal.graphics.nmr.rasterize_rgbad(faces, textures,
-                                                   self.image_size, self.anti_aliasing, self.near, self.far,
-                                                   self.rasterizer_eps, self.bg_color)
-            return out['rgb'], out['depth'], out['alpha']
-
-        # Render RGB channels only
-        elif mode == 'rgb':
-            images = kal.graphics.nmr.rasterize(faces, textures,
-                                                self.image_size, self.anti_aliasing, self.near, self.far,
-                                                self.rasterizer_eps, self.background_color)
-            return images
-
-        # Render depth image
-        elif mode == 'depth':
-            images = kal.graphics.nmr.rasterize_silhouettes(faces,
-                                                            self.image_size, self.anti_aliasing)
-
-        # Render only a silhouette, without RGB colors
-        elif mode == 'silhouette':
-            depth = kal.graphics.nmr.rasterize_depth(faces,
-                                                     self.image_size, self.anti_aliasing)
-            return depth
-
-        else:
-            raise ValueError('Mode {0} not implemented.'.format(mode))
-
-    def look_at(vertices, eye, at=torch.FloatTensor([0, 0, 0]),
+        face_vertices = self.vertices_to_faces(vertices, faces)
+        face_textures = self.vertices_to_faces(textures, faces)
```

Comment:
Is this normal to call `vertices_to_faces` with a textures as argument ?

Reply:
Yes, `vertices_to_faces` expands the vertices (or any similar vertex property for that matter) into a layout as defined by the faces. It is used for `textures` (and vertex colors, etc. if needed).